app/Resources/doc/import.md

```diff
@@ -53,30 +38,65 @@ case_ID,site,adm_date,...
 ...
 ```
 
-Would import the first and second rows and skip the third one, it would however notify you of a detected duplicate so you could 
-investigate if it truly is a duplicate.
+...would import the first and second rows and skip the third one, it would however notify you of a detected duplicate so you could investigate if it truly is a duplicate.
 
-The two fields are also used to find pre-existing cases to update. So using the same data example above. If case 1234, AGO-1 already
-exists in the database, a new record isn't created, but the existing record is retrieved and any fields in the file are updated.
+The two fields are also used to find pre-existing cases to update. So using the same data example above. If case 1234, AGO-1 already exists in the database, a new record isn't created, but the existing record is retrieved and any fields in the file are updated.
 
-The second option is the 'Case Id and Verify Country' option. This is designed primarily to handle importing RRL data that doesn't
-contain a proper site id. This linker will look for a case with the particular case id within a single country. If there is more 
-than one it will not import the data. This makes the assumption that in the surveillance network case id generation is at least 
-unique within each country.
+The second option is the 'Case Id and Verify Country' option. This is designed primarily to handle importing RRL data that doesn't contain a proper site id. This linker will look for a case with the particular case id within a single country. If there is more than one it will not import the data. This makes the assumption that in the surveillance network case id generation is at least unique within each country.
 
 Pre-Processors
 --------------
+
+Each column to be imported can have an associated **Pre-Processor** with one or more conditions to handle situations where the data in the import file is not an exact match to what should be entered in the case record. Pre-Processors can be fairly complex and are therefore quite flexible.
+
+![Edit Pre-Processor Conditions](images/editPreprocessorConditions.png)
+
+Rules and rule groups can be added and combined with AND and OR operators. The values of multiple input fields can be evaluated to determine the value to output to the case record.
+
+Once a Pre-Processor has been attached to a column, you will see **Modify** and **Delete** buttons for editing or removing the pre-processor.
+
 DB Column
-------------
+---------
+
+**DB Column** is the field in the case data to which the imported column data should be assigned. The selectable options are displayed as internal field names and types.
+
```

Comment:
It should also be noted that this and the validator field allows for typing to change/search. As oppose to having to scroll through. The search looks anywhere within the text. So a field like siteLab.csf_wcc you could find by typing wcc or csf. etc.